### ISSUE TITLE
prov/verbs;ofi_rxm: Are counters supported?

### ISSUE THREAD

**Issue by tonyzinger** [ofiwg/libfabric#4236]
When using the verbs;ofi_rxm provider, the counter values are not incremented after using the fi_read() or fi_write() APIs.

For example is some debug output from a fi_write() test:
[cn1, Rank:  0] DEBUG: sft_reset_counter, fi_cntr_read(): current cntr: 0
  :
  :
[cn1, Rank:  0] DEBUG: do_api: FI_WRITE(1): dt: UINT32, xfer: 1, flags: 0x0000000000000000, len: 8, user cntx: 0x1000001, ep: 0x2535330, local: 0x258dbc0, num: 1, fi_addr: 0x1, remote: 0x1800b80, dest_mr_key: 0x4cebf
[cn1, Rank:  0] DEBUG: do_api: FI_WRITE(1): xfer: 1, user cntx: 0x1000001, Successful
[cn1, Rank:  0] INFO: fi_cq_read(FI_WRITE), entries: 1, flags: 0x204, buffer: (nil), length: 0, data: 0x0, context: 0x1000001, tag: 0x7fa02cee5000
[cn1, Rank:  0] DEBUG: sft_get_cq_events: num_cq_requests: 1, current_cq: 1, cq_index: 0, uc: 0x1000001
[cn1, Rank:  0] DEBUG: do_transfers: FI_WRITE: memory_registration index: buf_index_for_rank: 0, cur_send_to: 1, mr_send_to: 1
[cn1, Rank:  0] DEBUG: do_api: FI_WRITE(1): dt: UINT32, xfer: 2, flags: 0x0000000000000000, len: 8, user cntx: 0x1000002, ep: 0x2535330, local: 0x258dbc8, num: 1, fi_addr: 0x1, remote: 0x1800b88, dest_mr_key: 0x4cebf
[cn1, Rank:  0] DEBUG: do_api: FI_WRITE(1): xfer: 2, user cntx: 0x1000002, Successful
[cn1, Rank:  0] INFO: fi_cq_read(FI_WRITE), entries: 1, flags: 0x204, buffer: (nil), length: 0, data: 0x0, context: 0x1000002, tag: 0x7ffe41b5cfc0
[cn1, Rank:  0] DEBUG: sft_get_cq_events: num_cq_requests: 1, current_cq: 1, cq_index: 0, uc: 0x1000002
[cn1, Rank:  0] ERROR: ltu_wait_counter Line: 2739, fi_cntr_wait() cntr_tries: 0, current cntr: 0, readerr cntr: 0, threshold: 2 returned 110 (Connection timed out)

**Comment by dmitrygladkov** [comment]
@tonyzinger Could you please verify if PR #4554 fixes this issue or not?

**Comment by tonyzinger** [comment]
@dmitrygladkov I built PR #4554 and used it in my testing.  I executed a RMA, MESSAGE and a TAGGED MESSAGES tests.  For the RMA test the counter did not get incremented.  It was still 0.
For the MESSAGE and TAGGED MESSAGE tests, the counter was incremented for the number of transfers that I did.  However, this was only 1/2 of the value that I expected, since I had counters enabled for both FI_SEND and FI_RECV.

I am attaching test debug output from all 3 tests and not Libfabric trace output.  Hopefully, this will help you debug the problem.

[counter_problem.txt](https://github.com/ofiwg/libfabric/files/2567516/counter_problem.txt)
